ffi/capi/include/diplomat_runtime.h

```diff
@@ -0,0 +1,22 @@
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct DiplomatWriteable {
+    void* context;
+    char* buf;
+    size_t len;
+    size_t cap;
+    void (*flush)(struct DiplomatWriteable*);
+    bool (*grow)(struct DiplomatWriteable*, size_t);
+} DiplomatWriteable;
```

Comment:
question: It's a little strange to me that this has `Diplomat` in the name? Was this something you were planning to clean up?

Reply:
I think the naming idea here was to make it clear that this is a Diplomat-provided type, rather than one exposed by the user's FFI declarations. For C++, though, perhaps we could have a `Writeable` in a `diplomat` namespace.

Reply:
I think the plan here is for `diplomat::WriteableTraits` to be a thing, and yes, we should be reexporting things from namespaces in C++

Reply:
Ah ok, to me as a user of the ICU4X FFI, I'm not sure I need to know what `diplomat` is, and having it in the name might be a bit confusing, but I do see the value in having a clean separation between what comes directly from the FFI and what is being provided by `diplomat`.

Reply:
I would just treat it as a "weird library type", not very different from if a C++ library was pulling in another library like abseil.

We could add flags to the cpp tool so that you can tell it where all of these things go.